src/dev/flang/fe/SourceModule.java

```diff
@@ -915,6 +915,10 @@ else if (c._hasPost != null && c._hasPostThen == null)
               }
             if (visibleFor(existing, f.outer()))
               {
+                if (existing.isNonArgumentField())
```

Comment:
I think we whould be stricter here: `existing.isNonArgumentField() || f.isNonArgumentField()`. If we redefine an argument field by a field that is not an argument field, we run into strange situations where the same field is initialized twice, once via `existing` in the inheritance call and once va `f` by the field declaration.